clang/lib/3C/3CStats.cpp

```diff
@@ -129,4 +130,69 @@ void PerformanceStats::printPerformanceStats(llvm::raw_ostream &O,
     O << "NumUnCheckedRegions:" << NumUnCheckedRegions << "\n";
 
   }
+}
+
+// Record Checked/Unchecked regions.
+bool StatsRecorder::VisitCompoundStmt(clang::CompoundStmt *S) {
+  auto &PStats = Info->getPerfStats();
+  if (S != nullptr) {
+    auto PSL = PersistentSourceLoc::mkPSL(S, *Context);
+    if (PSL.valid() && canWrite(PSL.getFileName())) {
+      switch (S->getWrittenCheckedSpecifier()) {
+      case CSS_None:
+        // Do nothing
+        break;
+      case CSS_Unchecked:
+        PStats.incrementNumUnCheckedRegions();
+        break;
+      case CSS_Memory:
+      case CSS_Bounds:
+        PStats.incrementNumCheckedRegions();
+        break;
+      }
+    }
+  }
+  return true;
+}
+
+// Record itype declarations.
+bool StatsRecorder::VisitDecl(clang::Decl *D) {
+  auto &PStats = Info->getPerfStats();
+  if (D != nullptr) {
+    auto PSL = PersistentSourceLoc::mkPSL(D, *Context);
+    if (PSL.valid() && canWrite(PSL.getFileName())) {
+      if (DeclaratorDecl *DD = dyn_cast<DeclaratorDecl>(D)) {
+        if (DD->hasInteropTypeExpr()) {
```

Comment:
What if this is a function decl, with an itype on multiple arguments. Will this over-count? E.g., it will visit the function decl and the vardecls?

Also, why are you not also counting normal checked types here? If they are handled elsewhere, then I would have thought that this code could go elsewhere too.

Reply:
Normal Checked C types will be counted in the ProgramInfo.cpp. Because we should handle `**` and other edge cases.